`shlib/shcrypto` tests fail ~10% of the time

If you run `go test` within shlib/shcrypto, roughly 10% of the time, it fails with:
```
panic: math/big: buffer too small to fit value
mat/big.nat.bytes(...)
math/big.(*Int).FillBytes()
github.com/shutter-network/shutter/shlib/shcrypto.bigToScalar()
```

Tests that trigger this:
* `TestEonPublicKeyMatchesSecretKey` (in `generateP2`)
* `TestMessageMalleability` (in `makeKeys` → `generateP2`)
* `TestC1Malleability` (in `makeKeys` → `generateP2`)
* `TestRoundTrip` (in `makeKeys` → `generateP2`)

You can reproduce this easier by doing `go test -count 1000` inside shlib/shcrypto.

--- a/shlib/go.mod
+++ b/shlib/go.mod
@@ -6,7 +6,7 @@ require (
 	github.com/ethereum/go-ethereum v1.13.4
 	github.com/google/go-cmp v0.6.0
 	github.com/pkg/errors v0.9.1
-	github.com/supranational/blst v0.3.11
+	github.com/supranational/blst v0.3.12
 	golang.org/x/crypto v0.14.0
 	gotest.tools v2.2.0+incompatible
 	gotest.tools/v3 v3.5.1

--- a/shlib/shcrypto/encryption_test.go
+++ b/shlib/shcrypto/encryption_test.go
@@ -219,6 +219,7 @@ func makeKeys(t *testing.T) (*EonPublicKey, *EpochSecretKey, *EpochID) {
 	eonSecretKey := big.NewInt(0)
 	for i := 0; i < n; i++ {
 		eonSecretKey.Add(eonSecretKey, ps[i].Eval(big.NewInt(0)))
+		eonSecretKey.Mod(eonSecretKey, order)
 
 		ss := []*big.Int{}
 		for j := 0; j < n; j++ {

--- a/shlib/shcrypto/keys_test.go
+++ b/shlib/shcrypto/keys_test.go
@@ -165,7 +165,7 @@ func TestEonPublicKeyMatchesSecretKey(t *testing.T) {
 	esk := big.NewInt(0)
 	for _, p := range []*Polynomial{p1, p2, p3} {
 		esk = esk.Add(esk, p.Eval(big.NewInt(0)))
-		esk = esk.Mod(esk, order) // Ensure esk stays within the field modulus
+		esk = esk.Mod(esk, order)
 	}
 
 	epkExp := generateP2(esk)

--- a/shlib/shcrypto/keys_test.go
+++ b/shlib/shcrypto/keys_test.go
@@ -165,7 +165,9 @@ func TestEonPublicKeyMatchesSecretKey(t *testing.T) {
 	esk := big.NewInt(0)
 	for _, p := range []*Polynomial{p1, p2, p3} {
 		esk = esk.Add(esk, p.Eval(big.NewInt(0)))
+		esk = esk.Mod(esk, order) // Ensure esk stays within the field modulus
 	}
+
 	epkExp := generateP2(esk)
 
 	gammas := []*Gammas{p1.Gammas(), p2.Gammas(), p3.Gammas()}

